src/components/main/EditingNameField.tsx

```diff
@@ -1,45 +1,57 @@
-import React, { useState } from 'react';
+import { ChangeEvent, useState } from 'react';
 
 import CloseIcon from '@mui/icons-material/Close';
 import DoneIcon from '@mui/icons-material/Done';
-import { Button, Stack, TextField } from '@mui/material';
+import { IconButton, Stack, TextField } from '@mui/material';
 
 import { useAccountTranslation } from '@/config/i18n';
 
-type EditableNameProps = {
+const MIN_USERNAME_LENGTH = 3;
+const MAX_USERNAME_LENGTH = 30;
+
+type EditingUserNameFieldProps = {
   name: string;
   onSave: (newValue: string) => void;
   onCancel: () => void;
 };
+
 const verifyUsername = (username: string) => {
   if (username.trim() === '') {
     return 'USERNAME_EMPTY_ERROR';
   }
-  if (username.length < 3 || username.length > 30) {
+  if (
+    username.length < MIN_USERNAME_LENGTH ||
+    username.length > MAX_USERNAME_LENGTH
+  ) {
     return 'USERNAME_LENGTH_ERROR';
   }
   return null;
 };
-let errorMessage;
+
 const EditingUserNameField = ({
   name,
   onSave,
   onCancel,
-}: EditableNameProps): JSX.Element => {
+}: EditingUserNameFieldProps): JSX.Element => {
   const [newUserName, setNewUserName] = useState(name);
   const { t } = useAccountTranslation();
   const [error, setError] = useState<string | null>();
-  const handleChange = ({ target }: React.ChangeEvent<HTMLInputElement>) => {
+
+  const handleChange = ({ target }: ChangeEvent<HTMLInputElement>) => {
     const { value } = target;
     setNewUserName(value);
-    errorMessage = verifyUsername(value);
-
-    if (errorMessage) {
-      setError(errorMessage ? t(errorMessage) : null);
-    }
+    const errorMessage = verifyUsername(value);
+    setError(
+      errorMessage
+        ? t(errorMessage, {
+            MIN: MIN_USERNAME_LENGTH,
+            MAX: MAX_USERNAME_LENGTH,
```

Comment:
Our convention for interpolation keys is to use camelCase.
```suggestion
            min: MIN_USERNAME_LENGTH,
            max: MAX_USERNAME_LENGTH,
```